src/plugins/opensearch_dashboards_utils/common/state_containers/create_state_container.ts

```diff
@@ -47,12 +46,18 @@ const isProduction =
     ? process.env.NODE_ENV === 'production'
     : !process.env.NODE_ENV || process.env.NODE_ENV === 'production';
 
+const deepFreeze = (value: any) => {
+  if (value && typeof value === 'object' && !Object.isFrozen(value)) {
+    Object.freeze(value);
+    Object.getOwnPropertyNames(value).forEach((prop) => deepFreeze(value[prop]));
+  }
+  return value;
+};
+
```

Comment:
`deep-freeze-strict` handled freezing functions. Will we not be needing that?

Reply:
@AMoo-Miki I added new changes, reflecting freezing functions.